src/windows/ms2mit/ms2mit.c

```diff
@@ -39,20 +41,61 @@ xusage(void)
     exit(1);
 }
 
-void
-main(
-    int argc,
-    char *argv[]
-    )
+/* Return true if princ is not a cross-realm principal. */
+krb5_boolean
+is_local_tgt(krb5_principal princ)
 {
-    krb5_context kcontext;
+    return princ->length == 2 &&
+    data_eq_string(princ->data[0], KRB5_TGS_NAME) &&
+    data_eq(princ->realm, princ->data[1]);
+}
+
+/*
+ * Check if a ccache has any tickets.
+ */
+static krb5_error_code
+cc_has_tickets(krb5_context kcontext, krb5_ccache ccache, int *has_tickets)
+{
+    krb5_error_code code;
+    krb5_cc_cursor cursor;
+    krb5_creds creds;
+
+    code = krb5_cc_set_flags(kcontext, ccache, KRB5_TC_NOTICKET);
+    if (code)
+        return code;
+
+    code = krb5_cc_start_seq_get(kcontext, ccache, &cursor);
+    if (code)
+        return code;
+
+    while (!has_tickets) {
+        code = krb5_cc_next_cred(kcontext, ccache, &cursor, &creds);
+        if (code)
+            break;
+
+        if ((!krb5_is_config_principal(kcontext, creds.server)) &&
+            (creds.times.endtime - time(0) > 0))
```

Comment:
This is over-parenthesized (you don't need the parens around either arm of the && operator).  I also suggest pulling the expression time(0) out of the loop, assigning its value to a krb5_timestamp variable, and using "creds.times.endtime > now" rather than subtraction.